lasagne/layers/merge.py

```diff
@@ -326,9 +323,36 @@ def __init__(self, incomings, merge_function, cropping=None, **kwargs):
         super(ElemwiseMergeLayer, self).__init__(incomings, **kwargs)
         self.merge_function = merge_function
         self.cropping = cropping
+        self.broadcastable = None
 
     def get_output_shape_for(self, input_shapes):
         input_shapes = autocrop_array_shapes(input_shapes, self.cropping)
+
+        input_dims = [len(shp) for shp in input_shapes]
+        if not all(input_dim == input_dims[0] for input_dim in input_dims):
+            raise ValueError('Input dimensions must be the same but were %s' %
+                             ", ".join(map(str, input_shapes)))
```

Comment:
I thought I finished my comments(this and at a couple of more places). I came back to see the diff and realized it's not published on this page until I click the "submit review" button (after writing all the comments).